### PR TITLE
feat/ update code

### PR DESCRIPTION
**Pull request by Rybasher**

### CODE REVIEW

File: mirror-web-server/src/login-code/login-code.controller.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { FirebaseTokenAuthGuard } from '../auth/auth.guard'
+import { LoginCodeService } from './login-code.service'
+
+@Controller('login-code')
+export class LoginCodeController {
+  constructor(private readonly loginCodeService: LoginCodeService) {}
+
+  @Post('generate-login-code')
+  @FirebaseTokenAuthGuard()
+  async createGTLoginCode(
```
Comment (RevoluPowered) [inline mirror-web-server/src/login-code/login-code.controller.ts:11]: Remove GT name

File: mirror-web-server/src/login-code/login-code.service.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      spaceId: new ObjectId(spaceId),
+      loginCode: uniqueLoginCode
+    })
+    return await createdGTLoginCode.save()
+  }
+
+  public async getLoginCodeRecordByLoginCode(
+    loginCode: string
+  ): Promise<LoginCode> {
+    const gtLoginCodeRecord = await this.loginCodeModel
```
Comment (RevoluPowered) [inline mirror-web-server/src/login-code/login-code.service.ts:66]: Remove GT

File: mirror-web-server/src/asset/asset.controller.ts [hunk 3]
```diff
@@ -785,6 +788,50 @@ export class AssetController {
...
+      userId,
+      assetId
+    )
+  }
+
+  @Post('asset-list/price')
+  @FirebaseTokenAuthGuard()
+  public async getAssetsListPrice(@Body() data: GetAssetsPriceDto) {
+    return await this.assetService.getAssetsListPriceAdmin(data.assetsIds)
+  }
```
Comment (Kluskey) [inline mirror-web-server/src/asset/asset.controller.ts:818]: rm

File: mirror-web-server/src/asset/asset.service.ts [hunk 4]
```diff
@@ -1188,6 +1424,50 @@ export class AssetService {
...
+      if (enabledPurchaseOptions.length > 1) {
+        this.logger.warn({
+          message: `Asset has more than one enabled purchase option of type ONE_TIME`,
+          assetId: asset._id
+        })
+      }
+    })
+
+    return assetsPriceWithPurchaseOptionIds
+  }
```
Comment (Kluskey) [inline mirror-web-server/src/asset/asset.service.ts:1469]: rm

File: mirror-web-server/test/stubs/spaceVersion.model.stub.ts [hunk 5]
```diff
@@ -298,8 +298,8 @@ export const spaceVersion1 = {
       webAppTourHasSeenPageCreateListing: false,
       webAppTourHasSeenPageAssets: false,
       emailVerified: false,
-      email: 'testaccount+test1@themirror.space',
-      displayName: 'Test Name 1!',
+      email: 'jeff+test1@themirror.space',
+      displayName: 'Jeff Test 1!',
```
Comment (Kluskey) [inline mirror-web-server/test/stubs/spaceVersion.model.stub.ts:302]: Use testaccount, not Jeff

File: mirror-web-server/test/stubs/spaceVersion.model.stub.ts [hunk 6]
```diff
@@ -624,8 +624,8 @@ export const spaceVersion2ForSpace20WithActiveSpaceVersion = {
...
```
Comment (Kluskey) [inline mirror-web-server/test/stubs/spaceVersion.model.stub.ts:628]: Use testaccount, not Jeff

File: mirror-web-server/src/asset/asset.service.ts [hunk 7]
```diff
@@ -1343,6 +1627,196 @@ export class AssetService {
...
+    return usersWithAmount
+  }
+
+  public async getAssetByPurchaseOptionIdAdmin(
+    purchaseOptionId: PurchaseOptionId
+  ) {
+    return await this.assetModel
+      .findOne({ 'purchaseOptions._id': new ObjectId(purchaseOptionId) })
+      .exec()
+  }
```
Comment (Kluskey) [inline mirror-web-server/src/asset/asset.service.ts:1767]: rm marketplace features

File: mirror-web-server/src/cron/cron.service.ts [hunk 8]
```diff
@@ -26,4 +32,36 @@ export class CronService {
...
+        CronService.name
+      )
+    } catch (err) {
+      this.logger.error(
+        'Update User Profiles In Mixpanel Failed',
+        err.stack,
+        CronService.name
+      )
+    }
+  }
```
Comment (Kluskey) [inline mirror-web-server/src/cron/cron.service.ts:66]: rm

File: mirror-web-server/src/login-code/login-code.service.ts [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    if (!space) {
+      throw new BadRequestException('Space not found')
+    }
+
+    // check if login code is unique
+    while (await this.loginCodeModel.findOne({ loginCode: uniqueLoginCode })) {
+      uniqueLoginCode = this._generateLoginCode()
+    }
+
+    const createdGTLoginCode = new this.loginCodeModel({
```
Comment (Kluskey) [inline mirror-web-server/src/login-code/login-code.service.ts:54]: rm GT

File: mirror-web-server/src/mixpanel/mixpanel.service.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+              }
+
+              resolve()
+            }
+          )
+        }
+      )
+    })
+  }
+}
```
Comment (Kluskey) [inline mirror-web-server/src/mixpanel/mixpanel.service.ts:39]: rm

**Review by Kluskey — APPROVED** [review]

**Pull request merged**